router/http/src/main/scala/io/buoyant/router/http/ApplyHostForwardedHeader.scala

```diff
@@ -0,0 +1,56 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Fields, Request, Response, HeaderMap}
+import com.twitter.util.Future
+import java.net.{Inet4Address, Inet6Address, InetSocketAddress, SocketAddress, URI, URISyntaxException}
+import scala.collection.mutable
+import scala.util.Random
+import com.twitter.logging.Logger
+import org.apache.commons.lang.StringUtils
+
+/**
+ * Applies the Host value in the [Forwarded](https://tools.ietf.org/html/rfc7239) header to the Request's Host header.
+ * So the Request is sent with the actual host value.
+ */
+object ApplyHostForwardedHeader {
+
+  object filter extends SimpleFilter[Request, Response] {
+    private val log = Logger(getClass)
+
+    val ForwardedHeader = "Forwarded"
+    def apply(req: Request, svc: Service[Request, Response]): Future[Response] = {
+
+      replaceHostWithForwardedHostIfExists(req)
+
+      svc(req)
+    }
+
+    private def getForwardedHost(headerMap: HeaderMap): Option[String] = {
+      headerMap.get(ForwardedHeader)
+        .flatMap { f =>
+          f.split(";|,").toStream
+            .map(_.trim)
+            .find(x => StringUtils.startsWithIgnoreCase(x, "host="))
+        }
+        .map(fHost => fHost.toLowerCase().replace("host=", "").replaceAll("\"|'", ""))
```

Comment:
What do you think about CharMatcher.is().trimFrom, to only remove leading / trailing quotes?